### PR TITLE
support mutil rootviews(2 RootView in one activity)

### PR DESCRIPTION
**Pull request by xincubate**
if there are 2 "RootView" in one acitivity, the logic in "MSREventBridgeModule.findReactRootView" would be found the first "RootView"，so all the event which will send to the second "RootView" are wrong，becasue they are sended to the first "RootView"。。。

this issue can be fix while find all the "RootView" in "MSREventBridgeModule.findReactRootView" ,otherwise find the first "RootView"

### CODE REVIEW

**Comment by thorbenprimke** [comment]
@xincubate - This certainly would work but it emits the event to all the `RootView`s. 
Do you have a reference to the `rootView` and thus instead of providing the activity, you could pass in the `rootView` to a method named `emitEventForRootView`?

**Comment by maicki** [comment]
I agree with @thorbenprimke.

We should move to the same view based API we have on iOS and deprecated the activity based on.  It would be up to the consumer within an `Activity` to dispatch to whatever view she would like to dispatch the event within an `Activity`.

Instead of
```
public void emitEventForActivity(Activity activity, final String name, @Nullable WritableMap info)`
```

We would add a method:
```
public void emitEventForView(View view, final String name, @Nullable WritableMap info)`
```

**Comment by xincubate** [comment]
@maicki  , @thorbenprimke, thanks for your answers !  Yes，I have all the reference of the RootView s.  emmmm... , I think "emitEventForView" would be more appropriate.

**Comment by xincubate** [comment]
@maicki , I update this PR. remove the method named findAllRootView and add emitEventForView

File: android/src/main/java/net/mischneider/MSREventBridgeModule.java [hunk 1]
```diff
@@ -103,9 +92,9 @@ public String getName() {
   // Clean up on CatalystInstance destroyed
   @Override
   public void onCatalystInstanceDestroy() {
-      super.onCatalystInstanceDestroy();
-      LocalBroadcastManager localBroadcastManager = LocalBroadcastManager.getInstance(mReactContext);
-      localBroadcastManager.unregisterReceiver(mLocalBroadcastReceiver);
+    super.onCatalystInstanceDestroy();
```
Comment (thorbenprimke) [inline android/src/main/java/net/mischneider/MSREventBridgeModule.java]: Can you revert these? - just seems like an indention change.

File: android/src/main/java/net/mischneider/MSREventBridgeModule.java [hunk 2]
```diff
@@ -239,6 +228,44 @@ public void emitEventForActivity(Activity activity, final String name, @Nullable
     localBroadcastManager.sendBroadcast(customEvent);
   }
 
+  /**
+   * Emits and event to a an event subscriber within a component that lifes within the components tree
+   * managed by the passed view
+   * Example: EventBridgeModule.emitEventForView(view, (MSREventBridgeInstanceManagerProvider)this.getContext(), "eventName", data);
+   */
+  static public void emitEventForView(View view, MSREventBridgeInstanceManagerProvider instanceManagerProvider, final String name, @Nullable WritableMap info) {
```
Comment (thorbenprimke) [inline android/src/main/java/net/mischneider/MSREventBridgeModule.java]: Let's change that to `RootView rootView` to be explicit.

File: android/src/main/java/net/mischneider/MSREventBridgeModule.java [hunk 3]
```diff
@@ -239,6 +228,44 @@ public void emitEventForActivity(Activity activity, final String name, @Nullable
...
+    final MSREventBridgeModule module = MSREventBridgeModule.getEventBridgeModule(instanceManagerProvider);
+    module.emitEventForView(view, name, info);
+  }
+
+  /**
+   * Emits and event to a an event subscriber within a component that lifes within the components tree
+   * managed by the passed view
+   * Example: EventBridgeModule.emitEventForView(view, "eventName", data);
+   */
+  public void emitEventForView(View view, final String name, @Nullable WritableMap info)  {
```
Comment (thorbenprimke) [inline android/src/main/java/net/mischneider/MSREventBridgeModule.java]: Here as well and then you can remove the `instanceOf` check below.

File: android/src/main/java/net/mischneider/MSREventBridgeModule.java [hunk 4]
```diff
@@ -270,4 +297,4 @@ private View findRootView(ViewGroup parent) {
     return null;
   }
 
-}
+}
```
Comment (thorbenprimke) [inline android/src/main/java/net/mischneider/MSREventBridgeModule.java]: Nit: Please revert. Let's keep the newline.

**Comment by maicki** [comment]
Thanks again @xincubate for working on this. Closed via #37

**Pull request closed without merging**